third_party/jsign-nodejs/index.js

```diff
@@ -0,0 +1,74 @@
+import {spawn} from 'child_process';
+import {dirname, join} from 'path';
+import {fileURLToPath} from 'url';
+
+/**
+ * Get the Java arguments to invoke jsign.jar.
+ * @returns {string[]} An array of ['-jar', <.../jsign.jar>].
+ */
+function getJsignJarArgs() {
+  const scriptPath = fileURLToPath(import.meta.url);
+  const currDir = dirname(scriptPath);
+  const jsignPath = join(currDir, 'jsign/jsign-4.0.jar');
+  return ['-jar', jsignPath];
+}
+
+/**
+ * Push [argName, argValue] to args if argValue if not falsy.
+ * @param {string[]} args
+ * @param {string} argName
+ * @param {string?} argValue
+ */
+function pushArgIfSpecified(args, argName, argValue) {
+  if (argValue) {
+    return args.concat(argName, argValue);
+  }
+  return args;
+}
+
+/**
+ * Get the jsign arguments according to options parameter.
+ * @param {object} options An object which will be used to generate the arguments.
+ * @returns {string[]} An array containing the corresponding jsign arguments.
+ */
+function getJsignOptionsArgs(options) {
+  var args = [];
+  args = pushArgIfSpecified(args, '--storetype', options.storeType);
+  args = pushArgIfSpecified(args, '--keystore', options.keyStore);
+  args = pushArgIfSpecified(args, '--storepass', options.storePassword);
+  args = pushArgIfSpecified(args, '--alg', options.algorithm);
+  args = pushArgIfSpecified(args, '--tsaurl', options.timestampAuthority);
+  return args;
+}
+
+/**
+ * Run jsign.jar according to the corresponding options targeting fileToSign.
+ * @param {string} fileToSign The path string of a file to be signed.
+ * @param {object} options The options to be passed to jsign.
+ * @param {'PKCS11'|'PKCS12'|'GOOGLECLOUD'|null} options.storeType The type of options.keyStore.
+ * @param {string?} options.keyStore The cert file, SunPKCS11 config, or cloud keystore name.
+ * @param {string?} options.storePassword The password to access options.keyStore.
+ * @param {'SHA-1'|'SHA-256'|null} options.algorithm The signing algorithm.
+ * @param {string?} options.timestampAuthority The URL of the timestamping authority.
+ * @returns {Promise<number>} A promise containing the exit code of jsign.
+ */
+export default async function jsign(fileToSign, options) {
+  if (!options) {
+    throw new Error('options is required by jsign');
+  }
+  if (!fileToSign) {
+    throw new Error('fileToSign is required by jsign');
+  }
+
+  const jsignProc = spawn(
+      'java',
+      [
+        ...getJsignJarArgs(),
```

Comment:
This is more readable:
```suggestion
       '-jar', getJsignJarPath(),
```